CMakeLists.txt

```diff
@@ -1,36 +1,44 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.8)
 
 project(PID-autotune)
 
-FIND_PACKAGE(Qt4 REQUIRED)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+find_package(Qt5 COMPONENTS Core Widgets PrintSupport REQUIRED)
+include_directories(${Qt5Core_INCLUDE_DIRS})
+include_directories(${Qt5Widgets_INCLUDE_DIRS})
+include_directories(${Qt5PrintSupport_INCLUDE_DIRS})
+set(POSITION_INDEPENDENT_CODE ON)
 
 file(
-	GLOB
-	source_files
-	src/*.cpp
-	third_party/qcustomplot/*.cpp
+        GLOB
+        source_files
+        src/*.cpp
+        third_party/qcustomplot/*.cpp
 )
 
 file(
-	GLOB
-	header_files
-	src/*.hpp
-	third_party/qcustomplot/*.h
+        GLOB
+        header_files
+        src/*.hpp
+        third_party/qcustomplot/*.h
 )
 
-QT4_WRAP_CPP(header_files_moc ${header_files})
+#QT4_WRAP_CPP(header_files_moc ${header_files})
 
-INCLUDE(${QT_USE_FILE})
+#INCLUDE(${QT_USE_FILE})
 
-ADD_DEFINITIONS(${QT_DEFINITIONS})
+#ADD_DEFINITIONS(${QT_DEFINITIONS} -fPIC)
```

Comment:
Hi @gagikh 
Thanks a lot for your contribution. I think you can remove those commented lines.